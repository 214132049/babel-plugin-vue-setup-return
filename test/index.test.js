const babel = require('@babel/core');
const vueSetupReturn = require('../lib');

it('example1', function() {
  var example = `
    import {ref} from '@vue/composition-api'
    
    export default {
      name: 'myComponent',
      setup () {
        const __RE__name = ref('名字')
      }
    }
  `;
  const { code } = babel.transform(example, {
    plugins: [
      vueSetupReturn
    ]
  })
  expect(code).toMatchSnapshot();
});

it('example2', function() {
  var example = `
    import {ref} from '@vue/composition-api'
    
    export default {
      name: 'myComponent',
      setup: function () {
        const __RE__name = ref('名字')
        
        return {
          __RE__name
        }
      }
    }
  `;
  const { code } = babel.transform(example, {
    plugins: [
      vueSetupReturn
    ]
  })
  expect(code).toMatchSnapshot();
});

it('example3', function() {
  var example = `
    import {ref} from '@vue/composition-api'
    
    export default {
      name: 'myComponent',
      setup () {
        const __RE__name = ref('名字')
        
        return {
          name
        }
      }
    }
  `;
  const { code } = babel.transform(example, {
    plugins: [
      vueSetupReturn
    ]
  })
  expect(code).toMatchSnapshot();
});

it('example4', function() {
  var example = `
    import {ref} from '@vue/composition-api'
    
    export default {
      name: 'myComponent',
      setup () {
        const __RE__name = ref('名字')
        
        return {}
      }
    }
  `;
  const { code } = babel.transform(example, {
    plugins: [
      vueSetupReturn
    ]
  })
  expect(code).toMatchSnapshot();
});

it('example5', function() {
  var example = `
    import {ref} from '@vue/composition-api'
    
    export default {
      name: 'myComponent',
      setup () {
        const __RE__name = ref('名字')
        
        return
      }
    }
  `;
  const { code } = babel.transform(example, {
    plugins: [
      vueSetupReturn
    ]
  })
  expect(code).toMatchSnapshot();
});

it('example6', function() {
  var example = `
    import {ref} from '@vue/composition-api'
    
    export default {
      name: 'myComponent',
      setup () {
        const __RE__name = ref('名字')
        
        return () => <div>{ __RE__name }</div>
      }
    }
  `;
  const { code } = babel.transform(example, {
    plugins: [
      '@babel/plugin-syntax-jsx',
      vueSetupReturn
    ]
  })
  expect(code).toMatchSnapshot();
});

it('example7', function() {
  var example = `
    import {ref} from '@vue/composition-api'
    
    export default {
      name: 'myComponent',
      filters: {
        setup (__RE__name) {
          console.log(__RE__name)
        }
      },
      setup () {
        const __RE__name = ref('0')
        const __RE__push = computed(() => __RE__name.value)
        const __RE__add = function() {
          __RE__name.value += 1
        }
        return function () {
          h('div', null, push)
        }
      }
    }
  `;
  const { code } = babel.transform(example, {
    filename: 'index.js',
    presets: [
      '@vue/app'
    ],
    plugins: [
      vueSetupReturn
    ]
  })
  expect(code).toMatchSnapshot();
});

it('example8', function() {
  var example = `
    import {ref} from '@vue/composition-api'
    const options = {
      name: 'myComponent',
      filters: {
        setup (__RE__name) {
          console.log(__RE__name)
        }
      },
      setup () {
        const __RE__name = ref('名字')
        const __RE__push = computed(() => __RE__name.value)
      }
    }
    export default options
  `;
  const { code } = babel.transform(example, {
    plugins: [
      vueSetupReturn
    ]
  })
  expect(code).toMatchSnapshot();
});

it('example8', function() {
  var example = `
    import {ref} from '@vue/composition-api'
    const options = {
      name: 'myComponent',
      filters: {
        setup (_$$_name) {
          console.log(_$$_name)
        }
      },
      setup () {
        const _$$_name = ref('名字')
        const _$$_push = computed(() => _$$_name.value)
      }
    }
    export default options
  `;
  const { code } = babel.transform(example, {
    plugins: [
      [vueSetupReturn, {
        prefix: '_$$_'
      }]
    ]
  })
  expect(code).toMatchSnapshot();
});
