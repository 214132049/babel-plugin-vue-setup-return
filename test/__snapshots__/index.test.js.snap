// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example1 1`] = `
"import { ref } from '@vue/composition-api';
export default {
  name: 'myComponent',

  setup() {
    const name = ref('名字');
    return {
      name: name
    };
  }

};"
`;

exports[`example2 1`] = `
"import { ref } from '@vue/composition-api';
export default {
  name: 'myComponent',
  setup: function () {
    const name = ref('名字');
    return {
      name: name
    };
  }
};"
`;

exports[`example3 1`] = `
"import { ref } from '@vue/composition-api';
export default {
  name: 'myComponent',

  setup() {
    const name = ref('名字');
    return {
      name: name
    };
  }

};"
`;

exports[`example4 1`] = `
"import { ref } from '@vue/composition-api';
export default {
  name: 'myComponent',

  setup() {
    const name = ref('名字');
    return {
      name: name
    };
  }

};"
`;

exports[`example5 1`] = `
"import { ref } from '@vue/composition-api';
export default {
  name: 'myComponent',

  setup() {
    const name = ref('名字');
    return {
      name: name
    };
  }

};"
`;

exports[`example6 1`] = `
"import { ref } from '@vue/composition-api';
export default {
  name: 'myComponent',

  setup() {
    const name = ref('名字');
    return () => <div>{name}</div>;
  }

};"
`;

exports[`example7 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _compositionApi = require(\\"@vue/composition-api\\");

var _default = {
  name: 'myComponent',
  filters: {
    setup(__RE__name) {
      console.log(name);
      return {};
    }

  },

  setup() {
    const name = (0, _compositionApi.ref)('0');
    const push = computed(() => name.value);

    const add = function () {
      name.value += 1;
    };

    return function () {
      h('div', null, push);
    };
  }

};
exports.default = _default;"
`;

exports[`example8 1`] = `
"import { ref } from '@vue/composition-api';
const options = {
  name: 'myComponent',
  filters: {
    setup(__RE__name) {
      console.log(name);
      return {};
    }

  },

  setup() {
    const name = ref('名字');
    const push = computed(() => name.value);
    return {
      name: name,
      push: push
    };
  }

};
export default options;"
`;

exports[`example8 2`] = `
"import { ref } from '@vue/composition-api';
const options = {
  name: 'myComponent',
  filters: {
    setup(_$$_name) {
      console.log(name);
      return {};
    }

  },

  setup() {
    const name = ref('名字');
    const push = computed(() => name.value);
    return {
      name: name,
      push: push
    };
  }

};
export default options;"
`;
